specs/IFrameMemoryUsage.md

```diff
@@ -0,0 +1,306 @@
+                    InitializeFrameEventView(webviewFrame);
+                    // ...
+                    return S_OK;
+                })
+                .Get(),
+            NULL);
+        //! [ICoreWebView2:add_RendererProcessIdChanged]
+        webview2_18->add_RendererProcessIdChanged(
+            Callback<ICoreWebView2RendererProcessIdChangedEventHandler>(
+                [this](ICoreWebView2* sender, IUnknown* args) -> HRESULT
+                {
+                    wil::com_ptr<ICoreWebView2_18> webview2_18;
+                    sender->QueryInterface(IID_PPV_ARGS(&webview2_18));
+                    UINT32 processId;
+                    //! [ICoreWebView2:get_RendererProcessId]
+                    CHECK_FAILURE(webview2_18->get_RendererProcessId(&processId));
+                    // Do something when renderer process id has been changed. 
+                    // For instance, retrieve the updated memory usage information 
+                    // for that main frame. 
+                    return S_OK;
+                })
+                .Get(),
+            NULL);
+    }
+}
+
+void InitializeFrameEventView(
+    wil::com_ptr<ICoreWebView2Frame> webviewFrame) 
+{
+    auto frame5 = webviewFrame.try_query<ICoreWebView2Frame5>();
+    if (frame5)
+    {
+        //! [ICoreWebView2Frame:add_FrameCreated]
+        frame5->add_FrameCreated(
+            Callback<ICoreWebView2NestedFrameCreatedEventHandler>(
+                [this](
+                    ICoreWebView2Frame* sender,
+                    ICoreWebView2FrameCreatedEventArgs* args) -> HRESULT
+                {
+                    wil::com_ptr<ICoreWebView2Frame> webviewFrame;
+                    CHECK_FAILURE(args->get_Frame(&webviewFrame));
+                    InitializeFrameEventView(webviewFrame);
+                    auto frame5 = webviewFrame.try_query<ICoreWebView2Frame5>();
+                    if (frame5)
+                    {
+                        //! [ICoreWebView2Frame:get_RendererProcessId]
+                        UINT32 processId;
+                        CHECK_FAILURE(frame5->get_RendererProcessId(&processId));
+                        // Do something when renderer process id has been changed. 
+                        // For instance, retrieve the updated memory usage information 
+                        // for that iframe.  
+                    }
+                    return S_OK;
+                })
+                .Get(),
+            NULL);
+        //! [ICoreWebView2Frame:add_RendererProcessIdChanged]
+        frame5->add_RendererProcessIdChanged(
+            Callback<ICoreWebView2FrameRendererProcessIdChangedEventHandler>(
+                [this](
+                    ICoreWebView2Frame* sender,
+                    IUnknown* args) -> HRESULT 
+                {
+                    wil::com_ptr<ICoreWebView2Frame5> frame5;
+                    sender->QueryInterface(IID_PPV_ARGS(&frame5));
+                    if (frame5) 

```

Comment:
This QI definitely should not fail because we must have successfully queried ICoreWebView2Frame5 from this very frame in order to register the event handler in the first place.

Reply:
Yes, please apply similar change as previous comment.